apps/ios/LiveVideo/LiveVideo/Views/Stream/PresentationLayout/PresentationLayoutViewModel.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  Copyright (C) 2022 Twilio, Inc.
+//
+
+import TwilioVideo
+import Combine
+
+class PresentationLayoutViewModel: ObservableObject {
+    struct Presenter {
+        var identity: String = ""
+        var displayName: String = ""
+        var presentationTrack: VideoTrack?
+    }
+
+    @Published var isPresenting = false
+    @Published var presenter = Presenter()
+    @Published var dominantSpeaker = SpeakerVideoViewModel()
```

Comment:
It is a little odd to default these to basically empty values but I think it is the best option. SwiftUI binding does not play nice with optional chaining. And so if these properties are optional it creates all sorts of issues in the UI when I try to bind some property of a struct to some view, because I have to unwrap the struct to get to the property. So instead these properties are not optional but the structs may basically have empty data inside them. The UI should rely on `isPresenting` to determine if these other properties have data that should be displayed.

We can talk offline about other solutions I thought of as well.

Reply:
This is how I set up my UIKit view models usually anyways to prevent them from being optional.

Reply:
Good to know thanks!